test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -70,46 +70,89 @@ def test_pitch(self, frequency):
+
+        """
+        amplitude_mult = 20.
+        power_mult = 10.
         amin = 1e-10
         ref = 1.0
-        db_multiplier = math.log10(max(amin, ref))
-
-        # Waveform amplitude -> DB -> amplitude
-        multiplier = 20.
-        power = 0.5
-
-        db = F.amplitude_to_DB(torch.abs(x), multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
+        db_mult = math.log10(max(amin, ref))
 
         # Spectrogram amplitude -> DB -> amplitude
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
+        db = F.amplitude_to_DB(spec, amplitude_mult, amin, db_mult, top_db=None)
+        x2 = F.DB_to_amplitude(db, ref, 0.5)
 
         torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
 
-        # Waveform power -> DB -> power
-        multiplier = 10.
-        power = 1.
-
-        db = F.amplitude_to_DB(x, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
-
         # Spectrogram power -> DB -> power
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
+        db = F.amplitude_to_DB(spec, power_mult, amin, db_mult, top_db=None)
+        x2 = F.DB_to_amplitude(db, ref, 1.)
+
+        self.assertEqual(x2, spec)
+
+    def test_amplitude_to_DB_batch(self):
+        torch.manual_seed(0)
+        spec = torch.rand([2, 2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_3dims(self):
+        """Test on a spectrogram with no batch dimension."""
+        torch.manual_seed(0)
+        spec = torch.rand([2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_2dims(self):
+        """Test on a spectrogram with no batch or channel dimensions."""
+        torch.manual_seed(0)
+        spec = torch.rand([100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def _check_top_db(self, spec):
+        amplitude_mult = 20.
+        amin = 1e-10
+        ref = 1.0
+        db_mult = math.log10(max(amin, ref))
+        top_db = 40.
+
+        decibels = F.amplitude_to_DB(spec, amplitude_mult, amin,
+                                     db_mult, top_db=top_db)
+        below_limit = decibels < 6.0205
+        assert not below_limit.any(), (
+            "{} decibel values were below the expected cutoff:\n{}".format(
+                below_limit.sum().item(), decibels
+            )
+        )
 
-        torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
+    def test_top_db_batch(self):
+        torch.manual_seed(0)
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
```

Comment:
The decibel cutoff is derived from the smallest value in the spectrogram, so in order to hard-code the cutoff, the smallest value needs to be predictable. There's also a (tiny) chance that no values large enough to be clamped are generated, so I manually set the max.

I default to using a random tensor when possible because it adds entropy to the inputs. It's just a rule of thumb, I can certainly change it.

I didn't really like this structure either, it is confusing. I've rewritten it to scale each spectrogram separately to strictly match the range (0, 200), which seems more sensible. Let me know what you think.

Reply:
Thanks, the comments you added nicely explain the intention well. I think this is good.